### PR TITLE
Shell: Fix parse mistake in '&&' not being recursive

### PR DESCRIPTION
**Pull request by alimpfard**
This fix brought to you by ~~Raid: Shadow Legends~~ the sudden split of #2824

Try:
```sh
$ foo && bar && baz
```

### CODE REVIEW

**Comment by alimpfard** [comment]
And yet...this one turned into a mess as well...
:facepalm:

File: AK/StringView.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ Vector<StringView> StringView::split_view(const StringView& separator, bool keep
         view = view.substring_view_starting_after_substring(part_with_separator);
         maybe_separator_index = view.find_first_of(separator);
     }
-    if (keep_empty || !view.is_empty())
+    if (!view.is_empty())
```
Comment (awesomekling) [inline AK/StringView.cpp:98]: Does any of the other split functions have this same problem?
* `String::split()`
* `String::split_view()`
Also please add a test :)

Comment (alimpfard) [reply AK/StringView.cpp:98]: Thinking a little about this, I am a little lost on whether this is actually a correct behaviour, should `"axbx".split_view('x', true)` be `"a", "b"` or `"a", "b", ""`?

I can think of reasons of both being correct, so I instinctively reach for keeping it as-is...

Comment (awesomekling) [reply AK/StringView.cpp:98]: The idea with `keep_empty` is that no empty strings will appear in the output.

Comment (alimpfard) [reply AK/StringView.cpp:98]: Then I'll revert my change and just add some tests.

**Pull request merged**